sdk/core/System.ClientModel/src/ModelReaderWriter/ModelJsonConverter.cs

```diff
@@ -0,0 +1,57 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Diagnostics.CodeAnalysis;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+namespace System.ClientModel.Primitives
+{
+    /// <summary>
+    /// A generic converter which allows <see cref="JsonSerializer"/> to be able to write and read any models that implement <see cref="IJsonModel{T}"/>.
+    /// </summary>
+#if !NET5_0 // RequiresUnreferencedCode in net5.0 doesn't have AttributeTargets.Class as a target, but it was added in net6.0
```

Comment:
Why are we using a NET5_0 TFM in a library that only targets net6.0?

Reply:
This is what is needed to get it to compile because there was a breaking change here.  For context https://github.com/Azure/azure-sdk-for-net/pull/38010

Reply:
But that was when it was in Azure.Core, which targets net5.0.  What is the correct solution in ClientModel?

Reply:
Does it help if I say that I'm pushing to see our `net5.0` target get killed off in MQ along with our `netcoreapp2.1`?

Reply:
All this says is that these APIs will already exist after 5 so don't add them in.  I don't think it has anything to do with us use 5 or not.